src/SFML/Audio/SoundStream.cpp

```diff
@@ -494,8 +498,11 @@ void SoundStream::clearQueue()
 ////////////////////////////////////////////////////////////
 void SoundStream::launchStreamingThread(Status threadStartState)
 {
-    m_isStreaming = true;
-    m_threadStartState = threadStartState;
+    {
+        std::scoped_lock lock(m_threadMutex);
+        m_isStreaming = true;
+        m_threadStartState = threadStartState;
+    }
```

Comment:
Ditto.

Reply:
In a hypothetical case, the write to the variables might not be visible to the thread that is started below. There are no formal guarantees for this.